### PR TITLE
sharing permissions for user when admin enables and disables various default share permissions

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Includes tests when admin changes default share permissions from admin's sharing settings and :
-  share with user without modifying the share permissions in the sharing dialog
-  share with user modifying the share permissions in the sharing dialog
## Related Issue

https://github.com/owncloud/core/issues/35846

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 1]
```diff
@@ -408,6 +408,42 @@ public function setSharingPermissions(
 		}
 	}
 
+	public function checkSharingPermissions(
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: add PHPdoc

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 2]
```diff
@@ -75,6 +75,10 @@ class AdminSharingSettingsPage extends SharingSettingsPage {
 	protected $deleteTrustedServerBtnXpath = "//ul[@id='listOfTrustedServers']//li[contains(., '%s')]//span[contains(@class, 'icon-delete')]";
 	protected $trustedServerErrorMsgXpath = "//p[@id='ocFederationAddServer']//span[@class='msg error']";
 
+	protected $createSharePermissionXpath = "//label[contains(text(),'Create')]";
+	protected $changeSharePermissionXpath = "//label[contains(text(),'Change')]";
+	protected $deleteSharePermissionXpath = "//label[contains(text(),'Delete')]";
+	protected $shareSharePermissionXpath = "//p[@id='shareApiDefaultPermissionsSection']//label[contains(text(),'Share')]";
```
Comment (phil-davis) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: Maybe add `//p[@id='shareApiDefaultPermissionsSection']` to the other `SharePermissionXpath`

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 3]
```diff
@@ -590,6 +590,28 @@ public function theUserSetsTheSharingPermissionsOfForOnTheWebUI(
 		);
 	}
 
+	/**
+	 * @Then the following permissions is seen for :fileName in the sharing dialog for user :userName
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]:
```suggestion
	 * @Then the following permissions are seen for :fileName in the sharing dialog for user :userName
```

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 4]
```diff
@@ -75,6 +75,10 @@ class AdminSharingSettingsPage extends SharingSettingsPage {
 	protected $deleteTrustedServerBtnXpath = "//ul[@id='listOfTrustedServers']//li[contains(., '%s')]//span[contains(@class, 'icon-delete')]";
 	protected $trustedServerErrorMsgXpath = "//p[@id='ocFederationAddServer']//span[@class='msg error']";
 
+	protected $createSharePermissionXpath = "//p[@id='shareApiDefaultPermissionsSection']//label[contains(text(),'Create')]";
+	protected $changeSharePermissionXpath = "//p[@id='shareApiDefaultPermissionsSection']//label[contains(text(),'Change')]";
+	protected $deleteSharePermissionXpath = "//p[@id='shareApiDefaultPermissionsSection']//label[contains(text(),'Delete')]";
+	protected $shareSharePermissionXpath = "//p[@id='shareApiDefaultPermissionsSection']//label[contains(text(),'Share')]";
```
Comment (skshetry) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]: Dai, Can we use `name` attribute to get the checkbox? The apps (eg: 'richdocuments') can add their own permissions (a.k.a advanced permissions), so, it'd be great if we could just interact with the checkbox from it's `name` attribute.

Comment (HariBhandari07) [reply tests/acceptance/features/lib/AdminSharingSettingsPage.php]: every checkbox have the same name attribute except the share which doesn't have a name attribute. So, I don't think it's possible.

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 5]
```diff
@@ -541,3 +542,120 @@ Feature: Sharing files and folders with internal users
...
+      | share  | yes |
+    And the user re-logs in as "user1" using the webUI
+    And the user opens folder "simple-folder" using the webUI
+    Then the option to rename file "lorem.txt" should be available on the webUI
+    And the option to delete file "lorem.txt" should not be available on the webUI
+    And the option to upload file should be available on the webUI
+    When the user shares file "lorem.txt" with user "User Three" using the webUI
+    Then as "user3" file "lorem.txt" should exist
+
+  Scenario: Create share when admin disables delete in share permissions
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
  Scenario: Create share when admin disables change in share permissions
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 6]
```diff
@@ -541,3 +542,120 @@ Feature: Sharing files and folders with internal users
...
+      | share  | yes |
+    And the user re-logs in as "user1" using the webUI
+    And the user opens folder "simple-folder" using the webUI
+    Then the option to rename file "lorem.txt" should not be available on the webUI
+    And the option to upload file should be available on the webUI
+    When the user shares file "lorem.txt" with user "User Three" using the webUI
+    Then as "user3" file "lorem.txt" should exist
+    And the option to delete file "lorem.txt" should be available on the webUI
+
+  Scenario: Create share when admin disables delete in share permissions
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
  Scenario: Create share when admin disables create and share in share permissions
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 7]
```diff
@@ -520,6 +520,7 @@ Feature: Sharing files and folders with internal users
     And the user re-logs in as "user1" using the webUI
     And the user opens folder "simple-folder" using the webUI
     Then the option to rename file "lorem.txt" should be available on the webUI
+    Then it should not be possible to share file "lorem.txt" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    And it should not be possible to share file "lorem.txt" using the webUI
```

**Review by phil-davis — APPROVED** [review]

**Comment by HariBhandari07** [comment]
Backport on https://github.com/owncloud/core/pull/35929

**Pull request merged**